src/scenes/Lines/scenes/Editor/JourneyPatterns/Editor/ServiceJourneys/Editor/PassingTimesEditor/index.tsx

```diff
@@ -52,25 +75,38 @@ const PassingTimesEditor = (props: Props & StateProps) => {
     setValidPassingTimes(isValid);
   }, [setValidPassingTimes, isValid]);
 
-  const onFieldChange = (index: number, field: string, value: any) => {
-    const newPt = passingTimes[index]
-      ? passingTimes[index].withFieldChange(field, value)
-      : new PassingTime({ [field]: value });
-    const newPts = replaceElement(passingTimes, index, newPt);
-
-    onChange(newPts);
+  const onFieldChange = (
+    index: number,
+    field: keyof PassingTime,
+    value: PassingTime[keyof PassingTime]
+  ) => {
+    const passingTime = passingTimes[index].withFieldChange(field, value);
+    const { time, offset } = getPassingTimeKeys(passingTime);
+    const newPassingTime = new PassingTime({
+      departureTime: passingTime[time],
+      arrivalTime: passingTime[time],
+      departureDayOffset: passingTime[offset],
+      arrivalDayOffset: passingTime[offset]
+    });
+    const newPassingTimes = replaceElement(passingTimes, index, newPassingTime);
+
+    onChange(newPassingTimes);
   };
 
   const handleDayOffsetChange = (
     index: number,
-    field: string,
-    value: string
+    field: keyof PassingTime,
+    value: PassingTime[keyof PassingTime]
   ) => {
     const parsedValue = parseInt(value);
     onFieldChange(index, field, parsedValue !== 0 ? value : undefined);
   };
 
-  const getDayOffsetDropDown = (tpt: any, index: number, field: string) => (
+  const getDayOffsetDropdown = (
+    tpt: PassingTime,
```

Comment:
Bare en filleting, men tenker det burde være mulig å finne et bedre navn enn `tpt` (jeg måtte ihvertfall lete litt rundt i koden for å skjønne hva det var en forkortelse for).

Reply:
Ja, borde egentligen bara vara `passingTime` eller något. Tar en runde på sånt sen!